client/components/navigation/NavBar.jsx

```diff
@@ -1,29 +1,108 @@
+
+
+export class NavBar extends Component {
+  state = {
+    user: {} || this.props.user,
+  };
+
+  componentDidMount() {
+    this.props.getCurrentUser()
+      .then((res) => {
+        const { user } = res.data;
+        if (!user) {
+          toastr.error('You are not logged in!');
+          this.props.history.push('/');
+        }
+      })
+      .catch(err => toastr.error(err));
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.user) {
+      this.setState({ user: nextProps.user });
+    }
+  }
+
+  render() {
+    const { user: { name } } = this.state;
+    return (
+      <div className="container-fluid navbar-wrapper">
+        <nav className="navbar navbar-expand-lg navbar-light pushit-nav">
+          <a className="navbar-brand border border-warning rounded pushit-brand" href="/home">PushIt</a>
+          <button
+            className="navbar-toggler"
+            type="button"
+            data-toggle="collapse"
+            data-target="#navbarSupportedContent"
+            aria-controls="navbarSupportedContent"
+            aria-expanded="false"
+            aria-label="Toggle navigation"
+          >
+            <span className="navbar-toggler-icon" />
+          </button>
+
+          <div className="collapse navbar-collapse" id="navbarSupportedContent">
+            <ul className="navbar-nav ml-auto">
+              <li className="nav-item active">
+                <a className="nav-link" href="/">Home <span className="sr-only">(current)</span></a>
+              </li>
+              <li className="nav-item">
+                <a className="nav-link" href="#section">Link</a>
+              </li>
+              <li className="nav-item dropdown">
+                <a
+                  className="nav-link dropdown-toggle"
+                  href="#section"
+                  id="navbarDropdown"
+                  role="button"
+                  data-toggle="dropdown"
+                  aria-haspopup="true"
+                  aria-expanded="false"
+                >
+                  <i className="fa fa-user" aria-hidden="true" />
+                  { name }
+                </a>
+                <div className="dropdown-menu" aria-labelledby="navbarDropdown">
+                  <a className="dropdown-item" href="#section">Action</a>
+                  <a className="dropdown-item" href="#section">Another action</a>
+                </div>
+              </li>
+            </ul>
+          </div>
+        </nav>
       </div>
-    </nav>
-  </div>
-);
+    );
+  }
+}
+
+NavBar.propTypes = {
+  user: shape({
+    id: PropTypes.number,
+    name: PropTypes.string,
+    email: PropTypes.string,
+    createdAt: PropTypes.string,
+    updatedAt: PropTypes.string,
+  }),
+  history: PropTypes.object, // eslint-disable-line react/forbid-prop-types
+  getCurrentUser: PropTypes.func.isRequired,
+};
+
+NavBar.defaultProps = {
+  user: {},
+  history: {},
+};
+
+const mapStateToProps = ({ user }) => {
```

Comment:
`const mapStateToProps = ({ user }) => ({ user })` will do here, see suggested changes to reducer below